shared/createHelper.js

```diff
@@ -224,9 +224,16 @@ function actuallyCreateApp(forcecli, config) {
         // Adding version
         config.version = SDK.version;
         
-        // Adding template repo uri and path if none provided
-        config.templaterepouri = config.templaterepouri || SDK.templatesRepoUri;
-        config.templatepath = config.templatepath || (config.templaterepouri == SDK.templatesRepoUri ? forcecli.appTypesToPath[config.apptype] : '');
+        // Figuring out template repo uri and path
+        if (config.templaterepouri) {
```

Comment:
If you specified a `templaterepouri`, we will "parse" it to figure out the repo url (to be used with the git clone command) and the path (into which to expect the template).

For instance
` createwithtemplate --templaterepouri=https://github.com/forcedotcom/SalesforceMobileSDK-Templates/iOSNativeTemplate#dev` 

is equivalent to 5.x's
 ` createwithtemplate --templaterepouri=https://github.com/forcedotcom/SalesforceMobileSDK-Templates#dev -templatepath=iOSNativeTemplate` 

(Of course that's the same as doing ` create --apptype=native` assuming `templatesRepoUri` in constants.js is `https://github.com/forcedotcom/SalesforceMobileSDK-Templates#dev`).